crates/rolldown/src/utils/normalize_options.rs

```diff
@@ -63,7 +63,9 @@ pub fn normalize_options(mut raw_options: crate::BundlerOptions) -> NormalizeOpt
     sourcemap: raw_options.sourcemap.unwrap_or(SourceMapType::Hidden),
     sourcemap_ignore_list: raw_options.sourcemap_ignore_list,
     sourcemap_path_transform: raw_options.sourcemap_path_transform,
-    shim_missing_exports: raw_options.shim_missing_exports.unwrap_or(false),
+    shim_missing_exports: raw_options.shim_missing_exports.unwrap_or_else(|| {
```

Comment:
This is not ideal and will cause enabling `shim_missing_exports` implicitly on all modues.

Reply:
https://github.com/rolldown/rolldown/blob/aef45e09fa93e7df8f18422fb8d5c47fae0c77f0/crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs#L446

Should check it here.

Probably also need to add a field in `NormalModule` to store `ModuleType`.